### ISSUE TITLE
Output lint errors that were caused by changes even if they're on lines that weren't changed

### ISSUE THREAD

**Issue by isaac-io** [AtakamaLLC/lint-diffs#21]
Thanks for this great tool! I use it daily to lint a large repository containing source code in many programming languages, and it's been immensely useful for me.

I've stumbled across a case where even though there aren't any errors in the lines that were changed, the changes do introduce new lint errors, but these aren't reported.

Consider the following Python script:

```python
import sys

print("Hello", file=sys.stderr)
```

It is later changed to read as follows:

```python
import sys

print("Hello")
```

flake8 reports an error in line 1, but lint-diffs ignores it because it's not in the lines that were changed:
```
$ flake8 foo.py
foo.py:1:1: F401 'sys' imported but unused
```

I realise that lint-diffs currently has no way to know which lint error was caused by the changes due to the fact that it only lints the current version of the file and ignores errors that are on lines outside the diff area. Perhaps it could also lint the version of the file before the change and compare the results, reporting newly introduced lint errors?

**Comment by earonesty** [comment]
We add this to our config:

`always_report=(E.*|.*unused-import)`

It will pick up on the unused imports, or any other error you deem "always" needs to be fixed.

**Comment by isaac-io** [comment]
Sure, that will get this specific situation fixed, but as I mentioned in my previous comment, in my repository I run a large number of linters, so tracking each and every error that can be introduced by a change on other lines for every linter is not really feasible.

It's also not really about errors that "always" need to be fixed, but about errors that the new changes introduced. I planned on replacing the myriad of scripts that we have for each and every language and technology with a single lint-diffs config in our CI, but I can't do that with the current behaviour, as it would mean that PRs might accidentally introduce new issues without us noticing it.

**Comment by earonesty** [comment]
that is an inherent problem with running lint on diffs, instead of running lint on everything.   lint-diffs is for gradually improving massive repos that need a lot of work to get "up to perfect" ... once they're perfect, just lint everything.

you could, instead, do this:  `always_report=(.*)`

that will report all linter errors for every changed file.   and basically ignore the diffs.  which might be what you want?   this would simplify your scripts, report all errors, and would still be more efficient

**Comment by earonesty** [comment]
ok, i've seen a few requests for "ignoring diffs", which seems to make the name of the tool a bit silly, but still useful for running many linters on changed files, or even many linters on all files.   will make cli options, so you don't have to add `always_report=(.*)` to everything
